pkg/controller/localvolumeset/localvolumeset_controller.go

```diff
@@ -100,54 +113,460 @@ func (r *ReconcileLocalVolumeSet) Reconcile(request reconcile.Request) (reconcil
+	ds := newDiscoveryDaemonsetForCR(cr)
+	oldDs := &appsv1.DaemonSet{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: ds.Name, Namespace: ds.Namespace}, oldDs)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return r.client.Create(context.TODO(), ds)
 		}
-
-		// Pod created successfully - don't requeue
-		return reconcile.Result{}, nil
-	} else if err != nil {
-		return reconcile.Result{}, err
+		return err
+	}
+	if !reflect.DeepEqual(ds.Spec, oldDs.Spec) {
+		return r.client.Update(context.TODO(), oldDs)
 	}
+	return nil
+}
 
-	// Pod already exists - don't requeue
-	reqLogger.Info("Skip reconcile: Pod already exists", "Pod.Namespace", found.Namespace, "Pod.Name", found.Name)
-	return reconcile.Result{}, nil
+func (r *ReconcileLocalVolumeSet) applyNodeSelector(cr *localv1alpha1.LocalVolumeSet, ds *appsv1.DaemonSet) *appsv1.DaemonSet {
+	nodeSelector := cr.Spec.NodeSelector
+	if nodeSelector != nil {
+		ds.Spec.Template.Spec.Affinity = &corev1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: nodeSelector,
+			},
+		}
+	}
+	return ds
 }
 
-// newPodForCR returns a busybox pod with the same name/namespace as the cr
-func newPodForCR(cr *localv1alpha1.LocalVolumeSet) *corev1.Pod {
+// newDiscDaemonsetForCR returns a busybox pod with the same name/namespace as the cr
+func newDiscoveryDaemonsetForCR(cr *localv1alpha1.LocalVolumeSet) *appsv1.DaemonSet {
 	labels := map[string]string{
 		"app": cr.Name,
 	}
-	return &corev1.Pod{
+	hostContainerPropagation := corev1.MountPropagationHostToContainer
+	volumes := []corev1.Volume{
+
+		{
+			Name: "local-disks",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: util.GetLocalDiskLocationPath(),
+				},
+			},
+		},
+	}
+	ds := &appsv1.DaemonSet{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      cr.Name + "-pod",
+			Name:      cr.Name + "-discovery",
 			Namespace: cr.Namespace,
 			Labels:    labels,
 		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: labels},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{Labels: labels},
+				Spec: corev1.PodSpec{
+					Volumes: volumes,
+					Containers: []corev1.Container{
+						{
+							Image:   "busybox",
+							Name:    "busybox",
+							Command: []string{"sleep", "3600"},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:             "local-disks",
+									MountPath:        util.GetLocalDiskLocationPath(),
+									MountPropagation: &hostContainerPropagation,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	addOwner(&ds.ObjectMeta, cr)
+	addOwnerLabels(&ds.ObjectMeta, cr)
+	return ds
+}
+
+// syncProvisionerConfigMap syncs the configmap and returns true if configmap was modified
+func (r *ReconcileLocalVolumeSet) syncProvisionerConfigMap(o *localv1alpha1.LocalVolumeSet) error {
```

Comment:
this does not handle update

Reply:
try https://godoc.org/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrUpdate

Reply:
updated